shardingsphere-governance/shardingsphere-governance-core/src/main/java/org/apache/shardingsphere/governance/core/registry/process/subscriber/ProcessRegistrySubscriber.java

```diff
@@ -68,8 +69,20 @@ public void loadShowProcessListData(final ShowProcessListRequestEvent event) {
      */
     @Subscribe
     public void reportExecuteProcessSummary(final ExecuteProcessSummaryReportEvent event) {
-        ExecuteProcessContext executeProcessContext = event.getExecuteProcessContext();
-        repository.persist(ProcessNode.getExecutionPath(executeProcessContext.getExecutionID()), YamlEngine.marshal(new YamlExecuteProcessContext(executeProcessContext)));
+        String executionID = event.getExecutionID();
+        synchronized (executionID) {
+            Map<String, Object> dataMap = event.getDataMap();
+            ExecuteProcessReportContext reportContext = (ExecuteProcessReportContext) dataMap.get(ExecuteProcessConstants.EXECUTE_ID.name());
+            if (null == reportContext) {
+                return;
+            }
+            YamlExecuteProcessContext yamlExecuteProcessContext = reportContext.getYamlExecuteProcessContext();
+            if (System.currentTimeMillis() - yamlExecuteProcessContext.getStartTimeMillis() <= reportContext.getShowProcessListNoReportThresholdMillis()) {
+                return;
+            }
+            repository.persist(ProcessNode.getExecutionPath(executionID), YamlEngine.marshal(yamlExecuteProcessContext));
+            reportContext.setReportToGovernanceDonePartially(true);
```

Comment:
Why do we need this `ReportToGovernanceDonePartially`?

Reply:
In general, it's used to indicate whether later report of the same sql need to persist to governance or not.

In details, there are 2 places will use `reportToGovernanceDonePartially`:
- In `reportExecuteProcessUnit` (sharding sqls execution in progress), if `total execution cost time not reach threshold && ! reportExecuteProcessUnit`, then it's not necessary to report to governance immediately, it might be reported later
- In `reportExecuteProcess` (execution completed), it need to delete process from governance, if there was no persistence to governance, then delete process from governance will be ignored for better performance